extensions/resource-metrics/resource-metrics-extention/ui/src/utils/api.ts

```diff
@@ -0,0 +1,51 @@
+export const apiCall = (url: string, headers: Record<string, string>) => {
```

Comment:
Extraction of logic looks good. Thank you!
My suggestion is rename this file to client.ts and put it together with the Metrics.tsx.

Reply:
Sure, Made the modification.